engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/text/AnyToDisplayTextNode.java

```diff
@@ -35,6 +35,21 @@ Text showExceptions(
     }
   }
 
+  @Specialization
```

Comment:
This is what I had in mind. You may need to add `@TruffleBoundary` - text concatenations aren't suitable for PE.